src/Bundle/spec/Sender/Adapter/SymfonyMailerAdapterSpec.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\MailerBundle\Sender\Adapter;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\Mailer\Event\EmailSendEvent;
+use Sylius\Component\Mailer\Model\EmailInterface;
+use Sylius\Component\Mailer\Renderer\RenderedEmail;
+use Sylius\Component\Mailer\Sender\Adapter\AbstractAdapter;
+use Sylius\Component\Mailer\SyliusMailerEvents;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\RawMessage;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+final class SymfonyMailerAdapterSpec extends ObjectBehavior
+{
+    function let(MailerInterface $mailer, EventDispatcherInterface $eventDispatcher): void
+    {
+        $this->beConstructedWith($mailer, $eventDispatcher);
+    }
+
+    function it_is_an_adapter(): void
+    {
+        $this->shouldHaveType(AbstractAdapter::class);
+    }
+
+    function it_sends_an_email(
+        MailerInterface $mailer,
+        EmailInterface $email,
+        EventDispatcherInterface $dispatcher,
+        RenderedEmail $renderedEmail
+    ): void {
+        $this->setEventDispatcher($dispatcher);
+
+        $renderedEmail->getSubject()->shouldBeCalled()->willReturn('subject');
```

Comment:
Usually `->shouldBeCalled()` is not needed when `->willReturn(...)` is called 🖖

Reply:
@Zales0123 Hum I do not agree. That allows to simplify specifications with only one sentence.

```php
$renderedEmail->getSubject()->willReturn('subject');

$renderedEmail->getSubject()->shouldBeCalled();
```

We already do that on resource bundle when it's easier to read.
https://github.com/Sylius/SyliusResourceBundle/blob/cc6381570bffa7d7e19e0899de5fa063a27abd79/src/Bundle/spec/Controller/FlashHelperSpec.php#L102